### ISSUE TITLE
`az acr login` does not work with `--subscription` argument

### ISSUE THREAD

**Issue by ppanyukov** [Azure/azure-cli#7649]
**Describe the bug**
`az acr login` fails if value of `--subscription` argument does not match active subscription as shown by `az account show`.

**To Reproduce**

Set active subscrption 

```
az account set --subscription SOME_SUB_NAME
```

Try `acr login` into another sub.
Observe "Access denied" message.
I then get prompted for username.

```
az acr login --name REG_NAME --resource-group utv --subscription REQUIRED_SUB_NAME
Unable to get AAD authorization tokens with message: Access to registry 'REG_NAME.azurecr.io' was denied. Response code: 401.
Username:
```

Set active sub to be the same as required target sub.
Observe everything works now.

```
az account set --subscription REQUIRED_SUB_NAME
az acr login --name REG_NAME --subscription REQUIRED_SUB_NAME
Login Succeeded
```

**Environment summary**
```
az --version
azure-cli (2.0.48)
```

**Comment by ehotinger** [comment]
@djyou @yuwaMSFT2

**Comment by djyou** [comment]
@ppanyukov `--subscription` specifies the subscription where your registry is in. It's used to login to a registry when your active subscription is not the one that has your registry.

**Comment by ppanyukov** [comment]
@djyou Yes, that's expected behaviour, except it doesn't work, please see the repro steps.

**Comment by djyou** [comment]
@ppanyukov The steps couldn't repro though.

When your active subscription is SOME_SUB_NAME:
```
az account set --subscription SOME_SUB_NAME
```

Were you able to show the registry?
```
az acr show --name REG_NAME --resource-group utv --subscription REQUIRED_SUB_NAME
```

If the `show` command works but `login` still doesn't work. Could you please share your registry name (or send to acrsup@microsoft.com)? Thanks.

**Comment by ppanyukov** [comment]
@djyou All very interesting because it definitely does not work for me.

Inicidentally, `az acr show` does work.  But `login` does not.

The registries are not only in different subscriptions, they are also in **different tenants**, with different AAD and everything.  Could this be the reason for `login` failure?

I will send registry and sub details over email, in case this helps.

**Comment by djyou** [comment]
Different tenants should be the cause. I will investigate and try to repro. Thanks.

**Comment by adewaleo** [comment]
I am working on a fix in #10364.

**Comment by kenorb** [comment]
I'm not sure, but I think this still doesn't work with the latest version.
I'm using SystemAssigned VM which is on Test subscription, and I'm trying to access ACR from Dev subscription (as per below).
I've assigned the right roles (_Contributor_ role for the whole subscription where is ACR) for that managed identity (VM), `az acr show` works fine, but not `az acr login`. So I think it's still related to that bug.

```
$ az login --identity -o table
EnvironmentName    HomeTenantId                          IsDefault    Name                          State    TenantId
-----------------  ------------------------------------  -----------  ----------------------------  -------  ------------------------------------
AzureCloud         a4be1f2e-....-....-....-............  True         Dev   						Enabled  a4be1f2e-....-....-....-............
AzureCloud         a4be1f2e-....-....-....-............  False        Test  						Enabled  a4be1f2e-....-....-....-............
$ az acr login --name testlzacr --subscription Test -o table
Login Succeeded
$ az acr show --name devlzacr -o table
NAME           RESOURCE GROUP      LOCATION    SKU      LOGIN SERVER              CREATION DATE         ADMIN ENABLED
-------------  ------------------  ----------  -------  ------------------------  --------------------  ---------------
devlzacr       ps-rg-.....-dev-lz  uksouth     Premium  devlzacr.azurecr.io  	  2024-05-02T11:43:16Z  True
$ az acr login --name devlzacr --subscription Dev -o table
Unable to get AAD authorization tokens with message: 2024-06-27 17:21:42.737838 An error occurred: CONNECTIVITY_REFRESH_TOKEN_ERROR
Access to registry 'devlzacr.azurecr.io' was denied. Response code: 403. Please try running 'az login' again to refresh permissions.
Error response from daemon: Get "https://devlzacr.azurecr.io/v2/": error parsing HTTP 403 response body: unexpected end of JSON input: ""
Login failed.
$ az acr repository list --name devlzacr -o table
Unable to get AAD authorization tokens with message: 2024-06-27 17:30:31.136494 An error occurred: CONNECTIVITY_REFRESH_TOKEN_ERROR
Access to registry 'devlzacr.azurecr.io' was denied. Response code: 403. Please try running 'az login' again to refresh permissions.
2024-06-27 17:30:41.538655 Error: requested access to the resource is denied. Correlation ID: 41aa55b3-bfe1-42a7-a07c-9443c81f11fe.
$ az version
{
  "azure-cli": "2.45.0",
  "azure-cli-core": "2.45.0",
  "azure-cli-telemetry": "1.0.8",
  "extensions": {
    "azure-devops": "0.26.0"
  }
}
```
